app/src/main/java/edu/wisc/ece/pinpoint/utils/FirebaseDriver.java

```diff
@@ -337,6 +354,33 @@ public Task<Map<String, Object>> fetchNearbyPins(@NonNull Location location) {
                 .continueWith(task -> (Map<String, Object>) task.getResult().getData());
     }
 
+    // TODO: improve fetch efficiency using query
+    public void fetchSocials() {
+        if (auth.getUid() == null) {
+            throw new IllegalStateException("User must be logged in to fetch followers");
+        }
+        followerIds = new HashSet<>();
+        followingIds = new HashSet<>();
+        db.collection("social").document(auth.getUid()).get().continueWith(task -> {
+            DocumentSnapshot doc = task.getResult();
+            ArrayList<String> followers = doc.get("followers") != null ? (ArrayList) doc.get("followers") :
+                    new ArrayList<>();
+            ArrayList<String> following = doc.get("following") != null ? (ArrayList) doc.get("following") :
+                    new ArrayList<>();
+            for(String person : followers){
+                followerIds.add(person);
+            }
+            for(String person : following){
+                followingIds.add(person);
+            }
+            db.collection("users").document(auth.getUid()).update("numFollowers",
+                    followerIds.size());
+            db.collection("users").document(auth.getUid()).update("numFollowing",
+                    followingIds.size());
```

Comment:
We shouldn't need to do this write here since we will update the user stats on follow/unfollow.